packages/fxa-settings/src/pages/Signup/Confirm/index.tsx

```diff
@@ -32,25 +59,128 @@ const Confirm = ({ email }: RouteComponentProps & ConfirmProps) => {
     instructionText: `Check your email for the confirmation link sent to ${email}`,
   };
 
-  const resendEmailHandler = async () => {
+  // Verifies if we have access to profile info and session token.
+  // If we do, set the email and render the page. If not, we don't know
+  // who the user is and can't confirm if we've sent a confirmation
+  // link, and we can't send a new one without an email address => redirect
+  // the user to retry signup.
+  useEffect(() => {
     try {
-      // TO-DO: signin confirmation email to user.
-      logViewEvent(viewName, 'resend', REACT_ENTRYPOINT);
-      setResendStatus(ResendStatus['sent']);
+      if (sessionTokenId) {
+        getEmail();
+      } else {
+        redirectToSignup();
+      }
     } catch (e) {
+      redirectToSignup();
+    }
+  }, [getEmail, sessionTokenId]);
+
+  useEffect(() => {
+    if (email) {
+      setShowLoadingSpinner(false);
+    }
+  }, [email, setShowLoadingSpinner]);
+
+  const navigateToConfirmCode = useCallback(
+    () =>
+      navigate('/confirm_signup_code?showReactApp=true', {
+        state: { email },
+        replace: true,
+      }),
+    [email, navigate]
+  );
+
+  // TODO implement broker to determine the the next screen
+  const navigateToNextScreen = () => {
+    logViewEvent(viewName, 'verification.success');
+    // TODO check if isForcePasswordChange
+    // navigate('/post_verify/password/force_password_change', {account})
+
+    // default behaviour: '/signup_confirmed'
+    // if Sync: '/connect_another_device'
+    // if Web: '/settings'
+    navigate('/settings', { replace: true });
+  };
+
+  // Adds a timeout before navigating to the /confirm_signup_code page
+  // when the user clicks on the resend email link.
+  useEffect(() => {
+    function onTimeout() {
+      navigateToConfirmCode();
+    }
+    let navigateTimeoutId: NodeJS.Timeout;
+    if (resendStatus === ResendStatus.sent) {
+      navigateTimeoutId = setTimeout(onTimeout, NAVIGATE_TIMEOUT);
+    }
+
+    return () => {
+      clearTimeout(navigateTimeoutId);
+    };
+  }, [navigateToConfirmCode, resendStatus]);
+
+  // Subscribe to account updates to find out if the session becomes verified.
+  // Stay on this page until the session is verified, then automatically
+  // navigate to the next screen.
+  useInterval(async () => {
+    try {
+      const sessionStatus = await account.isSessionVerified();
+      if (sessionStatus === true) {
+        navigateToNextScreen();
+        setIsPolling(null);
+      } else {
+        return;
+      }
+    } catch (e) {
+      setIsPolling(null);
+    }
+  }, isPolling);
+
+  function redirectToSignup() {
+    // TODO: Going from react page to non-react page will require a hard
+    // navigate. When signup flow has been fully converted we should be able
+    // to use `navigate`.
+    window.location.href = '/signup';
```

Comment:
Q: Should this be `window.location.href` or use reach router?
I was vaguely remembering @dschom's recent https://github.com/mozilla/fxa/pull/15087 PR.

Reply:
I think we still need to use `window.location.href` to navigate outside the React app until all routes are converted and available in React. @dschom can you confirm if your PR enables using navigate in this situation?